app/src/main/java/org/oppia/android/app/home/promotedlist/PromotedStoryListView.kt

```diff
@@ -24,37 +24,55 @@ class PromotedStoryListView @JvmOverloads constructor(
 
   @Inject
   lateinit var bindingInterface: ViewBindingShim
+  private var dataList: List<PromotedStoryViewModel> = listOf()
 
   override fun onAttachedToWindow() {
     super.onAttachedToWindow()
 
     (FragmentManager.findFragment<Fragment>(this) as ViewComponentFactory)
       .createViewComponent(this).inject(this)
 
-    adapter = BindableAdapter.SingleTypeBuilder.newBuilder<PromotedStoryViewModel>()
```

Comment:
Could you please add a test that repros the broken scenario & verify that it correctly breaks without the new fix in place? If getting the scrolling situation to repro the issue is too hard, consider just calling onAttachedToWindow directly in a way that's reasonably close to what Android would do in production. This might be the better approach since Robolectric doesn't simulate window-level operations very well.

Reply:
Does this mean mean mocking this custom view with the breaking implementation to reproduce the scenario? I don't think I understand what you mean by "correctly breaks"

Or do you mean verify that it _doesn't_ break?

Reply:
Oh sorry, what I mean is: please write a test that correctly tests this case (which means it should pass since the fix is currently checked in), and then manually revert just the fix part & re-run the test to verify it breaks (this is simulating the scenario where someone undoes your change--the test should catch that regression). This is a general best practice when writing tests since it's easy to sometimes write tests that won't ever fail (e.g. if they are setup incorrectly) which makes them unhelpful.

Reply:
Done

Reply:
Given that we're setting up the state very specifically, this is actually fine. Tests can make assumptions since they're responsible for arranging things in a specific way.